### DISCUSSION TITLE
fly hijack should search harder for a shell

### DISCUSSION THREAD

**Discussion by tremblaysimon** [concourse/concourse#5521] [General]
By default `fly hijack` command without command arg will try to start `bash` in the hijacked container. If the container doesn't contain `bash` it will produce an error.  So we will try to hijack with /bin/sh instead.

It's not a command that we use everyday and in Windows world (with git bash terminal for example) the command `fly -t your-target hijack -u build-url /bin/sh` will not work (we can use instead 
`fly -t your-target hijack -u build-url sh` and it will work...)

It should try to open `sh` if `bash` isn't found, at least. Maybe it should be configurable like it is mentioned in the old unresolved and closed issue https://github.com/concourse/concourse/issues/1073?

It's not a blocking issue, but it would be a nice feature to have. It would be convenient.

What do you think about that? Should we reopen https://github.com/concourse/concourse/issues/1073?

**Comment by vito** [comment]
This would be nice to have. 👍

I'm not going to re-open the issue because it's unlikely that we would prioritize this ourselves (our roadmap is pretty packed), but a PR for this would definitely be accepted, and I'm happy to help along the way.

Here's my comment from last time this came up that provides some guidance/context for anyone interested: https://github.com/concourse/concourse/pull/4755#issuecomment-556296316

**Reply by tremblaysimon** [reply]
Thank you I didn't see that old PR and your comment in it. Very interesting and and it will be a nice start point for me to contribute to Concourse!
